src/Illuminate/Notifications/BaseNotification.php

```diff
@@ -0,0 +1,43 @@
+<?php
+
+namespace Illuminate\Notifications;
+
+class BaseNotification
```

Comment:
Maybe we should make this an `abstract` class?

Reply:
TBH, I'd be tempted to make this into both a contract and a trait, so the implementing class `Notification` can pull in the trait.

Reply:
That's how I had it originally, but then changed it realising that the `Notification` class isn't and these would be used interchangeably. I've now changed it to abstract for now. 
I started with an interface, however we have 2 properties used in a few places hence I went with the `abstract` class rather than contract.

Reply:
@GrahamCampbell - any thoughts on it?